runtime/tchannel_outbound_call.go

```diff
@@ -89,15 +88,15 @@ func (c *tchannelOutboundCall) logFields(ctx context.Context) []zapcore.Field {
 		zap.Time("timestamp-finished", c.finishTime),
 	}
 
-	for k, v := range c.reqHeaders {
-		fields = append(fields, zap.String(
-			fmt.Sprintf("%s-%s", logFieldClientRequestHeaderPrefix, k), v,
-		))
-	}
-	for k, v := range c.resHeaders {
-		fields = append(fields, zap.String(
-			fmt.Sprintf("%s-%s", logFieldClientResponseHeaderPrefix, k), v,
-		))
+	var logFields []zap.Field
+	if c.client != nil && c.client.contextExtractor != nil {
+		headers := map[string]string{}
+		for k, v := range c.reqHeaders {
+			headers[k] = v
+		}
+		ctx = WithEndpointRequestHeadersField(ctx, headers)
+		logFields = append(logFields, c.client.contextExtractor.ExtractLogFields(ctx)...)
+		ctx = WithLogFields(ctx, logFields...)
```

Comment:
No need to put the log fields on to the context, the purpose of the method is to get log fields from context

Reply:
Done!